### PR TITLE
Adding support for setting spark job namespaces to all namespaces

### PR DESCRIPTION
**Pull request by ChenYi015**
## Purpose of this PR

Add support for setting `spark.jobNamespaces` to all namespaces (`""`).

- controller/webhook should watch all namespaces if `spark.jobNamespaces` is empty list or contains `""`
- should not create spark service account and rbac resources for empty namespace
- mutating/validating webhook configurations should not add `namespaceSelector` when `spark.jobNamespaces` is empty list or contains `""`

1. If we want to watch all namespaces, then we can install the spark operator as follows:

```bash
helm install spark-operator charts/spark-operator-chart \
    --create-namespace \
    --namespace spark-operator \
    --set 'spark.jobNamespaces={}'
```

but this will not create rbac resources for spark in any namespaces.

2. If we want to watch all namespaces and also create rbac resources for spark in namespace `default` and `spark-operaotr`, then we can do as follows:

```bash
helm install spark-operator charts/spark-operator-chart \
    --create-namespace \
    --namespace spark-operator \
    --set 'spark.jobNamespaces={,default,spark-operator}'
```

## Change Category
Indicate the type of change by marking the applicable boxes:

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [] Documentation update

## Checklist
Before submitting your PR, please review the following:

- [x] I have conducted a self-review of my own code.
- [ ] I have updated documentation accordingly.
- [x] I have added tests that prove my changes are effective or that my feature works.
- [ ] Existing unit tests pass locally with my changes.

### Additional Notes

If we set `spark.jobNamespaces` to `["","spark-operator"]`, then it will failes:

```bash
$ helm install spark-operator spark-operator/spark-operator \
    --version 2.0.0-rc.0 \
    --create-namespace \
    --namespace spark-operator \
    --set 'spark.jobNamespaces={,spark-operator}'
Error: INSTALLATION FAILED: 1 error occurred:
	* serviceaccounts "spark-operator-spark" already exists
```

### CODE REVIEW

**Comment by ChenYi015** [comment]
Since PR #2128 has already fixed the cache issue when the spark job namespaces contains namespaceAll (`""`), I will rebase it once that PR is merged.

**Comment by ChenYi015** [comment]
> #2128 is now merged. Do you want to update your PR?

@vara-bonthu Updated.

**Comment by vara-bonthu** [comment]
It would be nice to add these details to our docs as well

> If we want to watch all namespaces, then we can install the spark operator as follows:
> helm install spark-operator charts/spark-operator-chart \
>     --create-namespace \
>     --namespace spark-operator \
>     --set 'spark.jobNamespaces={}'
> but this will not create rbac resources for spark in any namespaces.
> 
> If we want to watch all namespaces and also create rbac resources for spark in namespace default and spark-operaotr, then we can do as follows:
> helm install spark-operator charts/spark-operator-chart \
>     --create-namespace \
>     --namespace spark-operator \
>     --set 'spark.jobNamespaces={,default,spark-operator}'
>

**Comment by ChenYi015** [comment]
@vara-bonthu Yes, I will update the docs once `v2.0.0` is released.

**Review by vara-bonthu — APPROVED** [review]
/approve

Will wait for another approval  @yuchaoran2011

**Review by yuchaoran2011 — APPROVED** [review]
/lgtm

**Pull request merged**
